### PR TITLE
Add support for Razer Huntsman

### LINKED ISSUE #739: Razer Huntsman 1532:0227
Hy there.

I want to use my Razer Huntsman keyboard Lighting functions. It's the cheaper version of Razer Huntsman Elite.
My version of keyboard, (lsusb : 1532:0227 Razer USA, Ltd ) isn't have memory for RGB program.
PLease tell me how can i use it with openrazer Debian Linux. The daemon is running correctly, but not recognize my device.
What can i do? Please help me with support.
[openrazer-daemon.log](https://github.com/openrazer/openrazer/files/2695998/openrazer-daemon.log)
 
Thanks

### PR DESCRIPTION
**Pull request by Zentetsu**
Based on @sumpster 's modification.

Fixes #739

### CODE REVIEW

**Comment by terricain** [comment]
You also need to update https://github.com/openrazer/openrazer/blob/master/install_files/udev/99-razer.rules

**Comment by z3ntu** [comment]
@haze-sama Did you test everything? Do all effects work properly?

**Comment by Zentetsu** [comment]
I only tested the color effects. 
The wave effect with left option doesn't work and I'm not sure, but it seems to me the slow and very slow option in reactive effect have the same duration.

**Comment by z3ntu** [comment]
Does "Wave right" actually show a wave effect that's going to the right?

**Comment by Zentetsu** [comment]
Yes, "waves right" works correctly.

**Comment by Nag1bat0r** [comment]
For me - work only color effect. Dont work Volume/Mute Next/Prev Track

**Comment by z3ntu** [comment]
Please rebase on master.

**Comment by tommyjcarpenter** [comment]
hey everyone, Is there a current barrier to merging this? Us hunstman's eagerly await! I globbed onto an issue (#739) today before I saw this, sorry!

**Comment by Nick-W** [comment]
Also eagerly awaiting this to merge, huntsman elite support being added to repo packages would be very, very much appreciated.

**Comment by tommyjcarpenter** [comment]
@Nick-W if I'm not mistaken this is only for the "regular" hunstman; huntsman elite PR was already closed I believe.

File: daemon/openrazer_daemon/hardware/keyboards.py
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/keyboards.py]: Please use https://assets.razerzone.com/eeimages/support/products/1360/1360_huntsman-3.png for this and the url below

File: driver/razerkbd_driver.h
Comment (z3ntu) [inline driver/razerkbd_driver.h]: Please fix this erroneous change

**Review by z3ntu — CHANGES_REQUESTED** [review]
See comments

**Comment by z3ntu** [comment]
Please remove the `.vscode/ipch/315cbeb247cb7fb3/mmap_address.bin` file

**Comment by z3ntu** [comment]
I'm still not sure if we should merge this with the wave effect to the left not working - or how we could fix that.

**Comment by ironLink21** [comment]
I've also been looking into this and I'm not able to get the Left Wave or Ripple to work.  I'm also not able to get anything with the FN key to work (volumn up/down/mute/kbd brightness(up/down)).  However you can control "brightness" & "gaming mode" via the client (polychromatic) so I'm wondering if it's just the FN key not being recognized which I've seen in the past with my laptop keyboard.
Running xev and watching the key presses those keys just come in as F1,F2,F3, etc.  When I run those same key combos from the laptop keyboard I get XF86AudioMute, XF86AudioLowerVolume, XF86AudioRaiseVolume.
Not sure how to get the actual key value for the FN key? That value is hard-coded in `razerkbd_driver.c:49` as `RAZER_FN 195` so maybe that's just a different value for the huntsman?

I agree with @z3ntu that this probably shouldn't get merged in just yet with those issues.

**Comment by tildearrow** [comment]
The FN key shouldn't generate any key codes, because the driver doesn't map it to any key. I'd like to see the ability to enable the FN key (https://github.com/openrazer/openrazer/pull/635#issuecomment-504566705) though.

**Comment by nightsky30** [comment]
> I'm still not sure if we should merge this with the wave effect to the left not working - or how we could fix that.

With this device using values 0,1 and chroma extended matrix wave for wave direction I believe that is the issue identified with https://github.com/openrazer/openrazer/pull/833#issuecomment-507472890

**Comment by fionera** [comment]
Any progress on this?

**Comment by cCurses** [comment]
Any progress on this?

**Comment by Woundorf** [comment]
What is required before merging this? Is something wrong still?

**Comment by pierrechopin** [comment]
Should I install this from sources, or should I really wait for the merge ?

**Comment by MarkusZoppelt** [comment]
> Should I install this from sources, or should I really wait for the merge ?

Interested in this as well. If there is no progress on this PR I will install from Zentetsu's source.

**Comment by craigccfl** [comment]
Please add another one to this list waiting for pid# 1532-0227  Razer Huntsman to be added to supported devices. Appreciate all the work done for the community.

**Review by z3ntu — APPROVED** [review]
LGTM, thanks and sorry for the delay!

If there are still minor issues (wave?), please open an issue for those and we'll see that we can fix those in the future.

**Pull request merged**
